### ISSUE TITLE
More than one octave

### ISSUE THREAD

**Issue by markusb** [cnlohr/colorchord#19]
I'd like to modify colorchord to cover more than 1 octave. As I understand it, it repeats itself for each octave right now. I'd like to change it so that is covers something like three octaves without repeating itself.

From the code it looks like I need something like FIXBPERO=72 and MAXNOTES=36 and would certainly have to replace ECCtoHEX to implement my own note to color translation.

Any thoughts on this ?

**Comment by cnlohr** [comment]
Though I believe this kind of defeats the point of ColorChord, I THINK the only change you would need to...

For CC2: make is to modify notefinder.c:230 That is where it folds the octaves.
For CCEmbedded: embeddednf.c:151. 

It would be a little jankey, since the term "octave" is confusing, but, should work?

**Comment by markusb** [comment]
Thanks Charles,

I'll start modding, I'll see where I get :-).

**Comment by cnlohr** [comment]
One big note:  I highly recommend messing with the desktop version first, then porting your changes to the embedded version.  It really is a lot easier (printf,etc.) to debug the desktop version.

**Comment by Ojsh** [comment]
Had some success doing a similar mod - pretty messy & ugly code changes at the moment, but can upload if it's of use to anyone. Wanted to light up a single LED per detected semitone across 4 octaves. - for application on a harp with 27 strings (110-1397Hz).  

Basically the changes were - as above starting at embeddednf.c:151 - to not fold DFT output into single folded_bins, instead keeping full data of FIXBINS size, then process most of the rest of that file to FIXBINS instead of FIXBPERO. Turned off first/last octave tapering, and burring for now (yet to revisit & test with included), also commented out areas where we're checking if have wrapped when determining distance between notes, as no longer relevant. Also changed SEMIBITSPERBIN to 0, and manually extended the range for 'thisfreq' using thisfreq = i*2; instead of shifting by semibitsperbin.  This works... though it could be two wrongs/three lefts make a right ; )

At the end of it all, have note_peak_freqs[] storing 8-bit int data over range 0-191, then use a custom lookup table to marry this up with desired output LED/string in embeddedout.c

Demo here:  https://youtu.be/B4XvRydglw0

**Comment by cnlohr** [comment]
Maybe I misunderstood.  It's not until the DFT folds that the octave information is ridded.  Without any modifications hardly at all you could get the data from all the octaves. The problem it looks like @markusb  was running into was that he wanted to change the fold-size.
